web/src/main/java/org/cbioportal/web/SampleController.java

```diff
@@ -15,45 +22,117 @@
-                                                                                       @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_SIZE) Integer pageSize,
-                                                                                       @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_NUMBER) Integer pageNumber,
-                                                                                       @PathVariable String studyId) {
+    @Autowired
+    private SampleService sampleService;
+
+    @RequestMapping(value = "/studies/{studyId}/samples", method = RequestMethod.GET,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation("Get all samples in a study")
+    public ResponseEntity<List<? extends SampleSummary>> getAllSamplesInStudy(
+            @ApiParam(required = true, value = "Study ID e.g. acc_tcga")
+            @PathVariable String studyId,
+            @ApiParam("Level of detail of the response")
+            @RequestParam(defaultValue = "SUMMARY") Projection projection,
+            @ApiParam("Page size of the result list")
+            @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_SIZE) Integer pageSize,
+            @ApiParam("Page number of the result list")
+            @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_NUMBER) Integer pageNumber,
+            @ApiParam("Name of the property that the result list is sorted by")
+            @RequestParam(required = false) SampleSortBy sortBy,
+            @ApiParam("Direction of the sort")
+            @RequestParam(defaultValue = "ASC") Direction direction) {
 
-        throw new UnsupportedOperationException();
+        if (projection == Projection.META) {
+            HttpHeaders responseHeaders = new HttpHeaders();
+            responseHeaders.add(HeaderKeyConstants.TOTAL_COUNT, sampleService.getMetaSamplesInStudy(studyId)
+                    .getTotalCount().toString());
+            return new ResponseEntity<>(responseHeaders, HttpStatus.OK);
+        } else {
+            return new ResponseEntity<List<? extends SampleSummary>>(
+                    sampleService.getAllSamplesInStudy(studyId, projection.name(), pageSize, pageNumber,
+                            sortBy == null ? null : sortBy.name(), direction.name()), HttpStatus.OK);
+        }
     }
 
-    @RequestMapping(value = "/studies/{studyId}/samples/{sampleId}", method = RequestMethod.GET)
+    @RequestMapping(value = "/studies/{studyId}/samples/{sampleId}", method = RequestMethod.GET,
+            produces = MediaType.APPLICATION_JSON_VALUE)
     @ApiOperation("Get a sample in a study")
-    public ResponseEntity<SampleSummary> getSampleInStudy(@PathVariable String studyId,
-                                                          @PathVariable String sampleId) {
+    public ResponseEntity<Sample> getSampleInStudy(
+            @ApiParam(required = true, value = "Study ID e.g. acc_tcga")
```

Comment:
nice to include examples. There may be a tag attribute for ApiParam to specify an example, but this is good.